### ISSUE TITLE
How to get frontmatter in v4 (client or server)

### ISSUE THREAD

**Issue by Firgrep** [shuding/nextra#3554]
**Is your feature request related to a problem? Please describe.** 
`useConfig()` handily exposed frontMatter, which allowed making components with conditions based on certain mdx file metadata.

**Describe the solution you'd like** 
A similar hook or function. I'm not sure if a replacement already exists but I can't find it.

**Comment by dimaMachina** [comment]
Hi, where do you want to access frontmatter in mdx page or custom jsx component imported in mdx page?

**Comment by Firgrep** [comment]
Custom JSX component that sits with the layout, just beneath the children, and renders its contents based on certain conditions from the frontmatter of the mdx page currently in view. E.g.

![image](https://github.com/user-attachments/assets/e454e773-daf3-4574-a52d-4778e27a705d)

I got this working in v3 making a wrapper component for `main` property in the old theme config, and inside the wrapper I had a component after the children that got the frontMatter from the old hook and rendered on certain conditions, something like: 

![image](https://github.com/user-attachments/assets/fcc08b26-606a-497d-b80b-39c3938af032)

**Comment by dimaMachina** [comment]
I think similar we return `activeThemeContext` https://github.com/shuding/nextra/blob/d5cb1bc0e84494cbcc8decbae56fb091ce9999f0/packages/nextra-theme-docs/src/stores/config.tsx#L79

In the same way, we can return `activeMetadata`

**Comment by Firgrep** [comment]
Yeah, that'd be amazing!

**Comment by dimaMachina** [comment]
Hi @Firgrep, I am finishing the Blogpost announcement for Nextra 4 RC.0, so busy for now to add it

I am planning to add this feat in Nextra 4 RC.1

For now, you can access `frontMatter` (in your client component) via:

```js
const { normalizePagesResult } = useConfig()
console.log(normalizePagesResult.activePath.at(-1).frontMatter)
```

Hope this unblock you for this moment ;)

**Comment by Firgrep** [comment]
Hi @dimaMachina , thanks for the hotfix! ;) Really appreciate it!

By the way, I've opened a new issue with regards to the bug, as I found another one and they seemed related. Ref #3567
